### PR TITLE
filer: support uploading file skip check expired entry

### PR DESCRIPTION
**Pull request by zhuqunzhou**
1.Like [2761](https://github.com/chrislusf/seaweedfs/pull/2761).The `skipCheckExpiredEntry` parameter also can reduce one `metadata` request per upload request.
2.fix the bug: Calculation error of method `skipCheckParentDirEntry`

### CODE REVIEW

**Comment by chrislusf** [comment]
the proto file change is not needed.

File: weed/server/filer_server_handlers_write_autochunk.go [hunk 1]
```diff
@@ -154,7 +158,7 @@ func (fs *FilerServer) saveMetaData(ctx context.Context, r *http.Request, fileNa
 			path += fileName
 		}
 	} else {
-		if fileName != "" {
+		if fileName != "" && !skipCheckExpiredEntry(r) {
```
Comment (chrislusf) [inline weed/server/filer_server_handlers_write_autochunk.go]: this is a wrong name. There are no expired entry here.

File: weed/pb/filer_pb/filer.pb.go [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.26.0
```
Comment (chrislusf) [inline weed/pb/filer_pb/filer.pb.go]: no needed.

Comment (famosss) [reply weed/pb/filer_pb/filer.pb.go]: ok

**Comment by chrislusf** [comment]
I want to avoid having too many "skipXXX" options.

This can be avoided by uploading to "http://filer:port/path/to/dir/", instead of "http://filer:port/path/to/dir/file"?

**Comment by zhuqunzhou** [comment]
> I want to avoid having too many "skipXXX" options.
> 
> This extra lookup can be avoided by uploading to `http://filer:port/path/to/dir/`, instead of `http://filer:port/path/to/dir/file`?

Ok. Please `close` it.

**Pull request closed without merging**